HW_13/task1-3.py

```diff
@@ -0,0 +1,161 @@
+import json
+from time import localtime, strftime
+
+# # Make new .json with testing data. Run once.
+# phonebook_dict = {
+#     "+11111111": ["Tom", "Jones"],
+#     "+33333333": ["Bob", "Marley"],
+#     "+55555555": ["Alice", "Cooper"],
+#     "+44444444": ["Tom", "Jones"],
+#     "+77777777": ["Tom", "Cruise"],
+#     "+88888888": ["Tom and Jerry", "Pets"],
+#     "101":["q", "w"]
+# }
+#
+# with open('phonebook_dict.json', "x") as phonebook:
+#     json.dump(phonebook_dict, phonebook)
+
+
+phonebook_dict = {}
+json_string = {}
+
+with open('phonebook_dict.json', "rt") as phonebook_json:
+    phonebook_dict = json.load(phonebook_json)
+
+
+# ===== Custom exception
+class MyCustomException(Exception):
+    def __init__(self, message, error_code):
+        super().__init__(message)
+        with open('my_custom_exeption.log', "a") as exeption_log_file:
+            exeption_log_file.write(f"""{(message)} with code {error_code} excepted at {strftime("%H:%M:%S", localtime())}\n""")
+        self.error_code = error_code
+
+    def get_error_code(self):
+        return self.error_code
+
+
+# Decorator section
+def definition_running_time(func):
+    def wraper(*args):
+        func(*args)
+        with open('definition_running_time.log', "a") as definition_log_file:
+            definition_log_file.write(f"""{func.__name__} run at {strftime("%H:%M:%S", localtime())}\n""")
+    return wraper
+
+
+# Definition section
+@definition_running_time
+def stats_command():
+    print(f"Count of items: {len(phonebook_dict)}")
+    return
+
+@definition_running_time
+def add_command(user_input):
+    user_input_split = user_input.split()
+    try:
+        if user_input_split[0] in phonebook_dict:
+            print("Same number is present. Can't add this item.")
+            raise MyCustomException("Custom exception is occurred", 555)
+            return None
+        else:
+            try:
+                phonebook_dict[user_input_split[0]] = user_input_split[1], user_input_split[2]
+            except IndexError:  # 3 text block needed (phone, first, last name). More - ok.
+                print("All data needed. Please write it.")
+            else:
+                print("Add item complete.")
+                with open('phonebook_dict.json', "w") as phonebook:
+                    json.dump(phonebook_dict, phonebook)
```

Comment:
nit: writing to file can be moved to own function to avoid duplication of code and for more readability